resharper/src/resharper-unity/Rider/ReSpeller/UnityByExtensionFileFilter.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using JetBrains.ReSharper.Features.ReSpeller.Analyzers;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Rider.ReSpeller
+{
+    [Language(typeof (KnownLanguage))]
+    public class UnityByExtensionFileFilter : ITypoAnalyzerFileFilter
+    {
+        public bool ShouldSkipFile(IFile file)
+        {
+            var sourceFile = file.GetSourceFile();
+            var path = sourceFile != null ? sourceFile.ToProjectFile()?.Location : null;
```

Comment:
Artem advised to take `GetLocation` from source file instead of doing `ToProjectFile`